lib/pages/athlete/bloc/athlete_page_bloc.dart

```diff
@@ -1,53 +1,120 @@
+import 'package:ax_dapp/util/supported_sports.dart';
 import 'package:equatable/equatable.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:intl/intl.dart';
 
 part 'athlete_page_event.dart';
+
 part 'athlete_page_state.dart';
 
 class AthletePageBloc extends Bloc<AthletePageEvent, AthletePageState> {
-  AthletePageBloc({required this.repo}) : super(AthletePageState.initial()) {
+  AthletePageBloc({
+    required this.mlbRepo,
+    required this.nflRepo,
+    required this.athlete,
+  }) : super(AthletePageState.initial()) {
     on<OnPageRefresh>(_mapPageRefreshEventToState);
     on<OnGraphRefresh>(_mapGraphRefreshEventToState);
   }
+  final AthleteScoutModel athlete;
 
-  final MLBRepo repo;
+  final MLBRepo mlbRepo;
+  final NFLRepo nflRepo;
 
   Future<void> _mapPageRefreshEventToState(
     OnPageRefresh event,
     Emitter<AthletePageState> emit,
   ) async {
-    try {
-      final playerId = event.playerId;
-      emit(state.copyWith(status: BlocStatus.loading));
-      final now = DateTime.now();
-      final startDate = DateTime(now.year, now.month - 1, now.day);
-      final formattedDate = DateFormat('yyyy-MM-dd').format(now);
-      final formattedStartDate = DateFormat('yyyy-MM-dd').format(startDate);
-      final until = formattedDate;
-      final from = formattedStartDate;
-      final stats = await repo.getPlayerStatsHistory(
-        playerId,
-        from,
-        until,
-      );
-      final graphStats = stats.statHistory
-          .map(
-            (stat) => GraphData(
-              DateFormat('yyy-MM-dd').parse(stat.timeStamp),
-              stat.price * 1000,
+    switch (athlete.sport) {
+      case SupportedSport.all:
+        //TODO: Handle this case.
+        break;
+
+      case SupportedSport.NFL:
+        try {
+          final playerId = event.playerId;
+          emit(state.copyWith(status: BlocStatus.loading));
+          final now = DateTime.now();
+          //final startDate = DateTime(now.year, now.month - 10, now.day);
+          final formattedDate = DateFormat('yyyy-MM-dd').format(now);
+          //final formattedStartDate =
+          // DateFormat('yyyy-MM-dd').format(startDate);
+          final until = formattedDate;
+          const from = '2022-05-02';
+          final stats = await nflRepo.getPlayerStatsHistory(
+            playerId,
+            from,
+            until,
+          );
+          final graphStats = stats.statHistory
+              .map(
+                (stat) => GraphData(
+                  DateFormat('yyy-MM-dd').parse(stat.timeStamp),
+                  stat.price * 10000,
+                ),
+              )
+              .toList();
+          final seenDates = <DateTime>{};
+          final distinctPoints = graphStats
+              .where((element) => seenDates.add(element.date))
+              .toList();
+          emit(
+            state.copyWith(
+              stats: distinctPoints,
+              status: BlocStatus.success,
```

Comment:
I think there is a better way of doing this than repeating code through each switch statement